### PR TITLE
Auto correlation

### PR DESCRIPTION
**Pull request by jtschwar**
Hi, 
I wrote a quick python script for performing Auto-Correlation, a Data Transform, on a single component dataset.

### CODE REVIEW

File: tomviz/DataTransformMenu.cxx [hunk 1]
```diff
@@ -132,12 +133,14 @@ void DataTransformMenu::buildTransforms()
...
+  new AddPythonTransformReaction(laplaceFilterAction, "Laplace Sharpen",
                                  readInPythonScript("LaplaceFilter"));
-  new AddPythonTransformReaction(gaussianFilterAction, "Gaussian Filter",
-                                 readInPythonScript("GaussianFilter"), false,
-                                 false,
-                                 readInJSONDescription("GaussianFilter"));
+  new AddPythonTransformReaction(autoCorrelationAction,"Auto Correlation",
+                                 readInPythonScript("Auto_Correlation"));
+  new AddPythonTransformReaction(gaussianFilterAction, "Gaussian Blur",
+                                   readInPythonScript("GaussianFilter"), false,
```
Comment (cryos) [inline tomviz/DataTransformMenu.cxx]: Indentation looks off in these lines, should be inline with the g of gaussianFilterAction

**Comment by cryos** [comment]
@cjh1 https://travis-ci.org/OpenChemistry/tomviz/jobs/313185274 shows clang-format failing, but it exited with 0, and so looked like it had passed.

File: tomviz/CMakeLists.txt [hunk 2]
```diff
@@ -236,6 +236,7 @@ set(exec_sources
 
 set(python_files
   AddPoissonNoise.py
+  Auto_Correlation.py
```
Comment (cjh1) [inline tomviz/CMakeLists.txt]: It seems that all other operator file names are using CamelCase, should we be consistent? i.e. AutoCorrelation.py.

Comment (cryos) [reply tomviz/CMakeLists.txt]: Yes, we should, good catch

File: tomviz/python/Auto_Correlation.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+def transform_scalars(dataset):
+    """Apply auto correlation on a dataset."""
+    
+    from tomviz import utils
+    from scipy.fftpack import fftn, ifftn, fftshift
+    import numpy as np
+    
+    array = utils.get_array(dataset)
+    dim = []
```
Comment (cjh1) [inline tomviz/python/Auto_Correlation.py]: This is not necessary as ```dim``` is assigned at line 10.

File: tomviz/python/Auto_Correlation.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    dim = array.shape
+    
+    #Transform the dataset.
+    result = np.empty_like(array)
```
Comment (cjh1) [inline tomviz/python/Auto_Correlation.py]: What are we creating an empty array and then overriding it?

**Comment by cjh1** [comment]
@cryos I will take a look at the clang-format issue.

File: tomviz/CMakeLists.txt (hunk 2 above)
Comment (jtschwar) [reply tomviz/CMakeLists.txt]: I wasn't aware of the formatting or clangformat when I submitted this branch. Thank you for informing me.

File: tomviz/python/Auto_Correlation.py (hunk 3 above)
Comment (jtschwar) [reply tomviz/python/Auto_Correlation.py]: The command in the line after, array.shape, gives me a tuple when I want create a numpy array.

Comment (cjh1) [reply tomviz/python/Auto_Correlation.py]: Correct, so you can delete line 9.

**Comment by cquammen** [comment]
@jtschwar Please address the flake8-identified indentation issues in the Python. They are identified here starting around output line 672: https://travis-ci.org/OpenChemistry/tomviz/jobs/334722570

**Comment by jtschwar** [comment]
@cquammen I believed I fixed all the formatting issues in my python file.

**Comment by jtschwar** [comment]
@cquammen Hi, thanks for letting me know about the formatting issues in python. I made the changes and they should be good now.

**Comment by cquammen** [comment]
> @cquammen I believed I fixed all the formatting issues in my python file.

I agree, I don't see any of the white spaces flake8 is complaining about. I wonder why we are seeing those failures.

**Pull request closed without merging**